src/gtpV2Codec/ieClasses/pgwsApnLevelLoadControlInformationInCreateSessionResponse.cpp

```diff
@@ -131,11 +131,12 @@ decodePgwsApnLevelLoadControlInformationInCreateSessionResponse(MsgBuffer &buffe
                          PgwsApnLevelLoadControlInformationInCreateSessionResponseData 
                          &data, Uint16 length)
 {
-
+    Uint16 groupedIeBoundary = length + buffer.getCurrentIndex();
     bool rc = false;
     GtpV2IeHeader ieHeader;
     set<Uint16> mandatoryIeLocalList = mandatoryIeSet;
-    while (buffer.lengthLeft() > IE_HEADER_SIZE)
+    while ((buffer.lengthLeft() > IE_HEADER_SIZE) &&
```

Comment:
after seeing so much of repetition..i kind of wonder why this code was also not done through template ?? Parsing bearer context in given message ?

Reply:
Can you clarify this..